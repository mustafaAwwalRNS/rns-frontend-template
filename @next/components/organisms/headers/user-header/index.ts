export * from "./user-header.types";
export * from "./user-header.organism";
