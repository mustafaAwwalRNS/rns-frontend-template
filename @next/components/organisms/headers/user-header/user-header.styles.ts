import styled from "styled-components";

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  ${({
    theme: {
      header: { height, zIndex },
      spacing,
      colors: { surface },
      shadow,
    },
  }) => ({
    boxShadow: shadow(2),
    height,
    zIndex,
    paddingLeft: spacing(2),
    paddingRight: spacing(2),
    backgroundColor: surface,
  })};
`;

export const HeaderLogo = styled.div`
  ${({ theme: { spacing } }) => ({
    marginRight: spacing(6),
  })}
`;
