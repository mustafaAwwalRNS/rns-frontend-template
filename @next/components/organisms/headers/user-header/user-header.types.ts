export interface UserHeaderProps {
  currentActiveRoute: string;
}
