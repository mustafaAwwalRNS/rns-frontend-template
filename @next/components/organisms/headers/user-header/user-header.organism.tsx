import React from "react";
import { Grid, GridItem, Button } from "@components/atoms";
import { UserAvatar } from "@components/molecules";
import * as S from "./user-header.styles";
import { useSelectAuth, authActions } from "@store";
import { useDispatch } from "react-redux";

export const UserHeader = (): JSX.Element => {
  const {
    user: { username, profileImage },
  } = useSelectAuth();
  const dispatch = useDispatch();
  return (
    <S.Header role="banner">
      <S.HeaderLogo>logo</S.HeaderLogo>
      <Grid spacing={2}>
        <GridItem>
          <Button onClick={() => dispatch(authActions.logout())}>Logout</Button>
        </GridItem>
        <GridItem>
          <UserAvatar username={username} imageSrc={profileImage} />
        </GridItem>
      </Grid>
    </S.Header>
  );
};
