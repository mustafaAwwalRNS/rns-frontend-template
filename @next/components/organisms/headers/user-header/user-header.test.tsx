import React from "react";
import { render, screen } from "test-utils";

import { UserHeader } from "./user-header.organism";

const setup = () => {
  const utils = render(<UserHeader />);
  const header = screen.getByRole("banner");
  return { header, ...utils };
};

describe("<UserHeader />", () => {
  it("should render", () => {
    const { header } = setup();
    expect(header).toBeInTheDocument();
  });
});
