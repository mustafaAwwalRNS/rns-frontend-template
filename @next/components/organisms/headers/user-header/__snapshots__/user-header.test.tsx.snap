// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserHeader /> should match snapshot 1`] = `
.c6 {
  margin: 0;
  padding: 0;
  color: black;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: calc(-1 * 1rem);
}

.c3 {
  padding: 1rem;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  height: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  border: none;
  cursor: pointer;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: purple;
  color: white;
  border-radius: 0.5rem;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
}

.c8 {
  width: 50px;
  height: 50px;
  overflow: 50%;
  min-width: 50px;
  min-height: 50px;
  border-radius: 50%;
  object-fit: cover;
  object-position: center;
  box-shadow: 0 2px 6px rgba(0,0,0,0.12),0 2px 4px rgba(0,0,0,0.24);
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  min-width: 80px;
  border: 1px solid #ccc;
  border-radius: 2rem;
  padding-left: 0.5rem;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
  margin-right: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  box-shadow: 0 2px 6px rgba(0,0,0,0.12),0 2px 4px rgba(0,0,0,0.24);
  height: 100px;
  z-index: 10;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: white;
}

.c1 {
  margin-right: 3rem;
}

<header
  class="c0"
  role="banner"
>
  <div
    class="c1"
  >
    logo
  </div>
  <div
    class="c2"
    role="grid"
    spacing="2"
  >
    <div
      class="c3"
      role="gridcell"
      spacing="2"
    >
      <button
        class="c4"
      >
        Logout
      </button>
    </div>
    <div
      class="c3"
      role="gridcell"
      spacing="2"
    >
      <div
        class="c5"
      >
        <span
          class="c6 c7"
        >
          tester
        </span>
        <img
          alt="user profile"
          class="c8"
          src="http://test-image"
        />
      </div>
    </div>
  </div>
</header>
`;
